### DISCUSSION TITLE
Implement Authentication

### DISCUSSION THREAD

**Discussion by flamecopper** [facebook/docusaurus#10281] [General]
Hi there. I would like to check if there any example of authentication, implementation of Docusaurus itself.

**Comment by OzakIOne** [comment]
As far as I know there is no authentication implemented in Docusaurus officially

**Comment by flamecopper** [comment]
We are trying to implement SSO authentication. Do let me know if there is any kind of custom code.

**Reply by slorber** [reply]
Docusaurus does not provide any auth function out of the box, and is only responsible for building static files, not for hosting and securing access to those underlying static files.

If you need to support SSO, I'd suggest looking at features of your host provider.

For example, it's possible to implement SAML with Vercel, or use an Edge middleware to implement any access control over the static files.

https://vercel.com/docs/security/saml
https://vercel.com/docs/functions/edge-middleware
https://vercel.com/docs/security/access-control
